utils/fhirpath/fhir_path_processor.bal

```diff
@@ -103,6 +103,79 @@ public isolated function evaluateFhirPath(map<json> fhirResource, string fhirPat
 
 }
 
+# Select the resource elements from the given FHIR resource.
+#
+# + fhirPaths - FhirPath expressions for the required elements
+# + fhirResource - Requested fhir resource
+# + return - Returns FHIR resource with the selected elements
+public isolated function selectResourceElements(string[] fhirPaths, map<json> fhirResource) returns map<anydata>|error {
+    string resourceType = check fhirResource["resourceType"].ensureType();
+
+    map<anydata> elementMap = {
+        "resourceType": {},
+        "meta": {}
+    };
+
+    foreach string fhirPath in fhirPaths {
+        string[] fhirPathTokens = regexp:split(re `\\.`, fhirPath);
```

Comment:
if we use regex like 
```
regexp:split(re `\s*\\.\s*`, fhirPath.trim());
```

We can avoid calling trim again and again.
wdyt?

Reply:
Updated in https://github.com/ballerina-platform/module-ballerinax-health.fhir.r4/commit/69a7f951571b72854ced9dd73557a36a00ffe97d